wai-handler-hal-cdk/runtime/default.nix

```diff
@@ -2,28 +2,55 @@
 # for deployment. This expression produces a directory containing the
 # `bootstrap` executable: CDK is smart enough to zip it up for deployment.
 { sources ? import ../../nix/sources.nix { }
-, compiler-nix-name ? "ghc8105"
+, compiler-nix-name ? "ghc8107"
+, architecture ? "x86_64"
 }:
 let
-  haskellNix = import sources."haskell.nix" { };
-  pkgs = haskellNix.pkgs-unstable;
-  pkgsMusl = haskellNix.pkgs-unstable.pkgsCross.musl64;
+  # Use local nixpkgs for UPX compression and cleaning the source
+  # repository, so we don't distribute more builds than necessary.
+  pkgsLocal = (import sources."haskell.nix" { }).pkgs-unstable;
+
+  # Be explicit about the architecture of the system performing this
+  # build, so that we can distribute builds if necessary, and we don't
+  # get broken cross-build configurations (e.g., trying to build linux
+  # musl64 from Darwin breaks).
+  pkgsMusl =
+    {
+      x86_64 = (import sources."haskell.nix" {
+        system = "x86_64-linux";
+      }).pkgs-unstable.pkgsCross.musl64;
+
+      # Cross-compile from x86_64 to ARM. This seems to want to build
+      # a cross-GHC, as well as do some magic with qemu, both of which
+      # take a very long time to build.
+      #
+      # It also seems to be broken at the moment - after making it
```

Comment:
Its not clear why we have it if its broken? Is it expected that once its fixed upstream it starts working here without extra actions? Is this `pkgsMusl` unused and left there for future?

Reply:
Yes. It's worked in the past and will do so again.